src/shared/bus-unit-util.c

```diff
@@ -1415,6 +1418,33 @@ static int bus_append_execute_property(sd_bus_message *m, const char *field, con
                 return 1;
         }
 
+        if (streq(field, "RootHash")) {
+                _cleanup_free_ char *text = NULL;
+                _cleanup_free_ void *roothash_decoded = NULL;
+                size_t roothash_decoded_size = 0;
+
+                if (is_path(eq)) {
+                        if (!path_is_absolute(eq))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "RootHash= '%s' is a relative path, ignoring: %m", eq);
+
+                        /* We have the path to a roothash to load and decode, eg: RootHash=/foo/bar.roothash */
+                        return bus_append_string(m, "RootHashPath", eq);
+                }
+
+                /* We have a roothash to decode, eg: RootHash=012345789abcdef */
+                text = strdup(eq);
+                if (!text)
+                        return -ENOMEM;
```

Comment:
actually, we strdup() this here? you can operate with the string as it is here, no?

Reply:
Leftover from the previous version where file was read immediately - will remove